crates/jwt/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+use base64::{engine::general_purpose, Engine as _};
+use dids::{bearer::BearerDid, document::KeySelector};
+use jws::{sign_compact_jws, verify_compact_jws, JwsError, JwsHeader};
+use serde::{Deserialize, Serialize};
+
+#[derive(thiserror::Error, Debug, Clone, PartialEq)]
+pub enum JwtError {
+    #[error(transparent)]
+    JwsError(#[from] JwsError),
+    #[error("serialization error {0}")]
+    SerializationError(String),
+    #[error("deserialization error {0}")]
+    DeserializationError(String),
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Claims {
+    #[serde(rename = "iss", skip_serializing_if = "Option::is_none")]
+    pub issuer: Option<String>,
+    #[serde(rename = "sub", skip_serializing_if = "Option::is_none")]
+    pub subject: Option<String>,
+    #[serde(rename = "aud", skip_serializing_if = "Option::is_none")]
+    pub audience: Option<String>,
+    #[serde(rename = "exp", skip_serializing_if = "Option::is_none")]
+    pub expiration: Option<i64>,
+    #[serde(rename = "nbf", skip_serializing_if = "Option::is_none")]
+    pub not_before: Option<i64>,
+    #[serde(rename = "iat", skip_serializing_if = "Option::is_none")]
+    pub issued_at: Option<i64>,
+    #[serde(rename = "jti", skip_serializing_if = "Option::is_none")]
+    pub jti: Option<String>,
+}
+
+impl Claims {
+    pub fn new(
+        issuer: Option<String>,
+        subject: Option<String>,
+        audience: Option<String>,
+        expiration: Option<i64>,
+        not_before: Option<i64>,
+        issued_at: Option<i64>,
+        jti: Option<String>,
```

Comment:
prob fine to exclude for now but we will need x5c

Reply:
yup! will come to when needed